packages/nextjs/app/debug/_components/contract/ContractInput.tsx

```diff
@@ -58,6 +59,11 @@ export const ContractInput = ({
           setFormErrorMessage={setFormErrorMessage}
         />
       );
+    }
+
+    // we prio tuples here to avoid wrong input
+    else if (isCairoTuple(paramType.type)) {
+      return <InputBase {...inputProps} />;
```

Comment:
Shall we have TupleInput just like ArrayInput, I draft one:
```
export const TupleInput = ({
  abi,
  parentForm,
  setParentForm,
  parentStateObjectKey,
  abiParameter,
  setFormErrorMessage,
}: TupleProps) => {
  const [inputElements, setInputElements] = useState<{ type: string, value: string }[]>([]);

  useEffect(() => {
    const parsed = parseTupleTypes(abiParameter.type);
    if (parsed.length) {
      setInputElements(parsed.map(item => ({ type: item, value: "" })))
    }
  }, [abiParameter.type]);

  useEffect(() => {
    setParentForm({
      ...parentForm,
      [parentStateObjectKey]: `(${inputElements.map(ele => ele.value).join(",")})`
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [JSON.stringify(inputElements, replacer)]);

  return (
    <div className="collapse bg-base-200 border-2 border-secondary flex-col space-y-4 p-6">
      {
        inputElements.map((element, index) => (
          <ContractInput
            abi={abi}
            key={index}
            setForm={(
              nextInputRecipe:
                | Record<string, any>
                | ((arg: Record<string, any>) => Record<string, any>),
            ) => {
              let nextInputObject = typeof nextInputRecipe === "function" ? nextInputRecipe(parentForm!) : nextInputRecipe;

              setInputElements((currentInputArray: any[]) => {
                currentInputArray[index].value = nextInputObject?.[index] || ""
                return [...currentInputArray];
              });
            }}
            form={inputElements.reduce((acc, curr, index) => {
              acc[index] = curr.value;
              return acc;
            }, {} as { [key: number]: string })}
            stateObjectKey={`${index}`}
            paramType={
              {
                name: `${abiParameter.name}(${index})`,
                type: element.type,
              } as AbiParameter
            }
            setFormErrorMessage={setFormErrorMessage}
          />
        ))
      }
    </div>
  );
};
```
<img width="658" alt="image" src="https://github.com/user-attachments/assets/59e55f79-f1de-499e-9d0a-47692a0b922f">

Reply:
please propose a separate PR ser, thanks for the effort!

Reply:
sure, will do